- Có bao nhiêu cột, hàng trong dữ liệu?
    Tôi dự đoán kết quả sẽ là 15 cột, 1000+ hàng

- Có bao nhiêu người sống sót?
    Tôi dự đoán kết quả sẽ là 300 

- Trong số người sống sót có bao nhiêu nam và nữ?
    Tôi dự đoán kết quả sẽ là 100+ nam và 200+ nữ

- Tỉ lệ nam sống sót, tỉ lệ nữ sống sót?
    Tôi dự đoán kết quả sẽ là 36% nam và 65% nữ

- Có bao nhiêu class vé?
    Tôi dự đoán kết quả sẽ là 3 class vé

- Giá cé cao nhất và thấp nhất?
    Tôi dự đoán kết quả sẽ là 800USD và 100USD

- Người có tuổi lớn nhất và nhỏ nhất?
    Tôi dự đoán kết quả sẽ là 90+ và 10 tháng tuổi

- Người có tuổi lớn nhất sống sót và nhỏ nhất sống sót?
    Tôi dự đoán kết quả sẽ là 60 tuổi và 11 tuổi

- Gia đình đông nhất đi cùng nhau là bao nhiêu?
    Tôi dự đoán kết quả sẽ là 6 người

- Town nào lên tàu nhiều nhất?
    Tôi dự đoán kết quả sẽ là Southampton

2026-05-13 22:56:25

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns

In [2]:
df = sns.load_dataset('titanic')
df.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


Dataset Titanic được tải bằng hàm `sns.load_dataset('titanic')` từ thư viện Seaborn.  
Lệnh `df.head()` dùng để hiển thị 5 dòng đầu tiên của bảng dữ liệu để quan sát nhanh cấu trúc và các cột dữ liệu.  
Bảng dữ liệu gồm 15 cột chứa thông tin về hành khách như giới tính, tuổi, hạng vé, giá vé và trạng thái sống sót.

In [3]:
df.tail()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
886,0,2,male,27.0,0,0,13.00,S,Second,man,True,NaN,Southampton,no,True
887,1,1,female,19.0,0,0,30.00,S,First,woman,False,B,Southampton,yes,True
888,0,3,female,NaN,1,2,23.45,S,Third,woman,False,NaN,Southampton,no,False
889,1,1,male,26.0,0,0,30.00,C,First,man,True,C,Cherbourg,yes,True
890,0,3,male,32.0,0,0,7.75,Q,Third,man,True,NaN,Queenstown,no,True


Lệnh `df.tail()` được sử dụng để hiển thị 5 dòng cuối cùng của DataFrame Titanic.  
Qua kết quả có thể quan sát các bản ghi cuối của dữ liệu cùng đầy đủ 15 cột thông tin của hành khách.  
Hàm này thường được dùng để kiểm tra dữ liệu ở cuối bảng hoặc xác nhận dữ liệu đã được tải đầy đủ.

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    str     
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    str     
 8   class        891 non-null    category
 9   who          891 non-null    str     
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    str     
 13  alive        891 non-null    str     
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), str(5)
memory usage: 100.4 KB


Lệnh `df.info()` dùng để hiển thị thông tin tổng quan của DataFrame Titanic.  
Kết quả cho biết tập dữ liệu có 891 dòng và 15 cột, đồng thời hiển thị kiểu dữ liệu của từng cột như `int64`, `float64`, `bool` và `category`.  
Hàm giúp kiểm tra số lượng giá trị không bị thiếu (`non-null`) và dung lượng bộ nhớ mà dữ liệu sử dụng.

In [5]:
df.describe()

,survived,pclass,age,sibsp,parch,fare
count,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


Lệnh `df.describe()` dùng để thống kê mô tả các cột dữ liệu dạng số trong DataFrame Titanic.  
Kết quả hiển thị các thông tin quan trọng như số lượng dữ liệu (`count`), giá trị trung bình (`mean`), độ lệch chuẩn (`std`), giá trị nhỏ nhất (`min`) và lớn nhất (`max`).  
Hàm giúp phân tích nhanh đặc điểm dữ liệu và phát hiện các giá trị bất thường trong tập dữ liệu.

In [6]:
df.shape

(891, 15)

Lệnh `df.shape` dùng để cho biết DataFrame Titanic có bao nhiêu dòng và bao nhiêu cột.  
Kết quả trả về là `(891, 15)`, nghĩa là tập dữ liệu có 891 dòng và 15 cột `(rows, columns)`.  
Hàm này thường được sử dụng để kiểm tra kích thước tổng quát của dữ liệu trước khi phân tích.

In [7]:
miss = df.isnull().sum()
pct  = (miss / len(df) * 100).round(1)
report = pd.DataFrame({'count': miss, 'pct_%': pct})
report = report[report['count'] > 0].sort_values('pct_%', ascending=False)
report

,count,pct_%
deck,688,77.2
age,177,19.9
embarked,2,0.2
embark_town,2,0.2
